File: Chapitre_2/exo2_2.py
import os

fils_1 = os.fork()      #Creation de fils 1
if fils_1 == 0 :
    print("Je suis le fils 1")
    os._exit(0)
fils_2=os.fork()        #Creation de fils 2
if fils_2==0 :
    print("Je suis le fils 2")
elif fils_1 != 0 and fils_2 != 0 :
    print("Je suis le père")

File: Chapitre_2/exo2_2.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>

int main()
{
    pid_t fils_1,fils_2;
    fils_1=fork();      //Creation de fils 1
    if(fils_1==0) {
        printf("Je suis le fils 1\n");
        exit(0);
    }
    fils_2=fork();      //Creation de fils 2
    if (fils_2==0 ) {
        printf("Je suis le fils 2\n");
    }
    else if(fils_1!=0 && fils_2!=0) {
        printf("Je suis le père\n");
    }
    return(0);
}